python/activator/middleware_interface.py

```diff
@@ -809,6 +811,47 @@ def _export_ml_models(self):
             _log.debug("Found %d new ML model datasets.", len(models))
         return models
 
+    def _get_init_output_types(self, run):
+        """Identify the specific init-output types to query.
+
+        Parameters
+        ----------
+        run : `str`
+            The name of an output run of the pipeline of interest. Different
+            pipelines may produce different sets of init-outputs.
+
+        Returns
+        -------
+        init_types : collection [`str`]
+            The init-output types of interest to Prompt Processing.
+        """
+        # Need to distinguish init-outputs from past visits' outputs.
+        all_names = self.central_butler.collections.get_info(run, include_summary=True).dataset_types
+        all_types = {self.central_butler.registry.getDatasetType(name) for name in all_names}
+        return {t.name for t in all_types if not t.dimensions}
```

Comment:
FWIW, you can also get this from the `PipelineGraph`s a bit more directly but probably less succinctly; you'd have to iterate over the outputs of the task init nodes, and then add `packages` manually.  But I think your prescription should be correct for any pipeline PP could potentially run (i.e. where all non-init outputs must have `visit` in their dimensions).

Reply:
Hmm... maybe something to add in [DM-40245](https://rubinobs.atlassian.net/browse/DM-40245)?